internal/controller/secretsync_controller.go

```diff
@@ -106,10 +107,21 @@ type SecretSyncReconciler struct {
 //+kubebuilder:rbac:groups="",resources=events,verbs=create;patch
 //+kubebuilder:rbac:groups=secrets-store.csi.x-k8s.io,resources=secretproviderclasses,verbs=get;list;watch
 
-func (r *SecretSyncReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+func (r *SecretSyncReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Result, err error) {
+	begin := time.Now()
+
 	logger := log.FromContext(ctx)
 	logger.Info("Reconciling SecretSync", "namespace=", req.NamespacedName.String())
 
+	defer func() {
+		wasSuccessful := err == nil
+		if !wasSuccessful && strings.Contains(strings.ToLower(err.Error()), "validatingadmissionpolicy") {
+			r.MetricReporter.ReportVAPDenyMetric(ctx, err.Error(), req.Namespace)
```

Comment:
I am not that familiar with this repository. Why is ValidatingAdmissionPolicy deny any different from any other authorization failure?